### ISSUE TITLE
Tableau Desktop is using an expired refresh token when requesting a new access token

### ISSUE THREAD

**Issue by carlost-mtz** [tableau/wdc3#28]
Hello, I am currently developing a web data connector that uses OAuth 2.0 to authenticate the application on behalf of a user in a REST API. I am using Taco toolkit 2.0, Tableau Desktop 2023.3, and the WDC SDK 3.0.

Everything works fine during the authentication process, the application can retrieve an access token, and I can request information to the REST API. However, I noticed an issue after retrieving data. When I switch between **Data Source** and **Sheet 1** views, the browser is automatically started showing me the redirect page of Tableau, it seems that the authentication was completed again, but I was only switching views, this happens every time I switch to the **Data Source** view.

![Screenshot 2024-01-18 183839](https://github.com/tableau/wdc3/assets/126598927/1f836d14-fe0e-4d27-a346-5d9f28ef484e)

After doing a little bit of debugging using Fiddler, I noticed that Tableau is making a request for a new access token in the background, after retrieving data, multiple times, it seems that the first time is working well, but the second time Tableau is using the refresh token that was used in the first request, this is a problem for me because **the REST API** I’m working on **does not allow to use the same refresh token multiple times**, it can only be used once. The REST API will provide a new refresh token in each request.

I have made a quick diagram to show the current behavior of Tableau, and how it’s expected to work with my REST API.

![Diagram1](https://github.com/tableau/wdc3/assets/126598927/8c854c1e-7c06-49f7-bf1e-fc4de8458087)

In conclusion, it seems that Tableau is using an expired refresh token multiple times, and that is not valid for the REST API I’m working on, that’s why the browser is started asking for credentials again. So, I was wondering, is there a configuration that I’m missing to have the expected behavior in my connector? Or is this an issue in Tableau Desktop or the SDK?

**Comment by maazamalik** [comment]
Hey, Were you able to fix the issue ?
